pkg/lockfile/match-package-json_test.go

```diff
@@ -179,3 +179,162 @@ func TestPackageJSONMatcher_Match_TransitiveDependencies(t *testing.T) {
 		},
 	})
 }
+
+func TestPackageJSONMatcher_Match_NameConflict(t *testing.T) {
+	t.Parallel()
+
+	sourceFile, err := lockfile.OpenLocalDepFile("fixtures/package-json/name-conflict/package.json")
+	if err != nil {
+		t.Errorf("Got unexpected error: %v", err)
+	}
+
+	packages := []lockfile.PackageDetails{
+		{
+			Name:           "aws-sdk-client-mock",
+			TargetVersions: []string{"^2.1.1"},
+		},
+		{
+			Name:           "aws-sdk-client-mock-jest",
+			TargetVersions: []string{"^2.1.1"},
+		},
+	}
+	err = packageJSONMatcher.Match(sourceFile, packages)
+	if err != nil {
+		t.Errorf("Got unexpected error: %v", err)
+	}
+
+	expectPackages(t, packages, []lockfile.PackageDetails{
```

Comment:
This expect packages was written before osv used snaps, it may be a more readable test if we used snap here, could you check please ?

Reply:
Every other test in the `lockfile` module still uses this `expectPackages` method. Are you proposing to use snap just with the matchers?

Reply:
Well, what I suggest is to try to migrate to snaps to :
- avoid having to use manually written test helpers (mainly because snap is a well tested project whereas our helpers are not tested)
- shorten the length of tests functions
- increase the readability of both tests and expectations

With this PR you're creating a new test, that's why I'm suggesting to use snap for it. For other tests we should migrate them to snap when we need to update them (because of a bug, a new feature, etc...)